turbo/jsonrpc/receipts/bor_receipts_generator.go

```diff
@@ -0,0 +1,106 @@
+package receipts
+
+import (
+	"context"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/erigontech/erigon-lib/chain"
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/kv"
+	"github.com/erigontech/erigon-lib/kv/rawdbv3"
+	"github.com/erigontech/erigon/consensus"
+	"github.com/erigontech/erigon/core"
+	"github.com/erigontech/erigon/core/state"
+	"github.com/erigontech/erigon/core/types"
+	"github.com/erigontech/erigon/core/vm"
+	"github.com/erigontech/erigon/core/vm/evmtypes"
+	bortypes "github.com/erigontech/erigon/polygon/bor/types"
+	"github.com/erigontech/erigon/turbo/services"
+	"github.com/erigontech/erigon/turbo/shards"
+)
+
+type BorGenerator struct {
+	receiptCache *lru.Cache[libcommon.Hash, *types.Receipt]
+}
+
+func NewBorGenerator(cacheSize int) *BorGenerator {
+	receiptCache, err := lru.New[libcommon.Hash, *types.Receipt](cacheSize)
+	if err != nil {
+		panic(err)
+	}
+
+	return &BorGenerator{
+		receiptCache: receiptCache,
+	}
+}
+
+func (g *BorGenerator) GenerateBorReceipt(ctx context.Context, tx kv.Tx, block *types.Block, msgs []*types.Message, engine consensus.EngineReader, chainConfig *chain.Config, txNumsReader rawdbv3.TxNumsReader, headerReader services.HeaderReader, blockReceipts []*types.Receipt) (*types.Receipt, error) {
```

Comment:
Instead of creating a function with 9 parameters you can instantiate `BorGenerator` with `blockReader` and with `engine` as part of `NewBorGenerator` - analogous to `receipts.NewGenerator`. 

That will eliminate the need for your function to pass in 3 inputs:
- `engine`
- `headerReader`
- `txNumsReader` - can create internally like in `receipts.Generator`